src/main/java/io/knowledgeassets/myskills/server/notification/AbstractNotificationResponse.java

```diff
@@ -2,8 +2,10 @@
 
 import com.fasterxml.jackson.annotation.JsonSubTypes;
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
-import io.knowledgeassets.myskills.server.communityuser.registration.notification.InvitationToJoinCommunityNotificationResponse;
-import io.knowledgeassets.myskills.server.communityuser.registration.notification.RequestToJoinCommunityNotificationResponse;
+import io.knowledgeassets.myskills.server.communityuser.registration.notification.acceptance.AcceptanceToCommunityNotificationResponse;
```

Comment:
I found another way to make Jackson generate the type property for the child classes having control of the value Jackson writes into the type property.
There is no need in `@JsonSubTypes` annotation anymore.

Jackson uses the same `com.fasterxml.jackson.databind.jsontype.TypeSerializer` for the parent `io.knowledgeassets.myskills.server.notification.AbstractNotificationResponse` and for its child classes.
Here are a couple of links to Jackson internals that helped me to figure out how Jackson works with `@JsonTypeInfo`.
https://github.com/FasterXML/jackson-databind/blob/master/src/main/java/com/fasterxml/jackson/databind/ser/std/AsArraySerializerBase.java#L63
https://github.com/FasterXML/jackson-databind/blob/master/src/main/java/com/fasterxml/jackson/databind/jsontype/impl/TypeNameIdResolver.java#L102
https://github.com/FasterXML/jackson-databind/blob/master/src/main/java/com/fasterxml/jackson/databind/ser/std/CollectionSerializer.java#L132
(The links can help us to proceed with the research in case we need it)

`TypeSerializer` is constructed based on the parent's `@JsonTypeInfo`.
As a result there will be the type property both for the parent class and for its child classes.
And the permanent class-name independent value of the type property of child classes can be set with `com.fasterxml.jackson.annotation.JsonTypeName` annotation.

Reply:
This is great. 👍